core/src/main/java/org/mule/runtime/core/internal/type/catalog/ModuleTypesUtils.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.internal.type.catalog;
+
+import static java.lang.String.format;
+import static java.util.stream.Collectors.joining;
+import static org.mule.metadata.api.utils.MetadataTypeUtils.getTypeId;
+import static org.mule.runtime.extension.api.util.ExtensionMetadataTypeUtils.getAlias;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import org.mule.metadata.api.model.MetadataType;
+import org.mule.metadata.message.api.el.ModuleDefinition;
+import org.mule.runtime.api.meta.model.ExtensionModel;
+
+/**
+ * Util class to retrieve a {@link MetadataType} from {@link ExtensionModel}
+ * 
+ * @since 4.5
+ */
+public class ModuleTypesUtils {
+
+  /**
+   * Retrieves a {@link MetadataType}
+   *
+   *
+   * @param moduleDefinitions the list of {@link ModuleDefinition} that may contain the {@link MetadataType}.
+   * @param moduleName        the identifier for the module where the {@link MetadataType} is defined.
+   * @param typeIdentifier    either the typeId or the typeAlias for the {@link MetadataType}.
+   *
+   * @return the {@link MetadataType}
+   */
+  public static MetadataType getType(Collection<ModuleDefinition> moduleDefinitions,
```

Comment:
I don't see much value on this being in an utils file. THis should be part of the ArtifactTypeLoader and be cached